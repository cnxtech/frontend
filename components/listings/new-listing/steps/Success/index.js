import React, { PureComponent } from 'react'
import Router from 'next/router'
import routerEvents from 'next-router-events'

import Row from '@emcasa/ui-dom/components/Row'
import Col from '@emcasa/ui-dom/components/Col'
import Icon from '@emcasa/ui-dom/components/Icon'
import Text from '@emcasa/ui-dom/components/Text'
import Button from '@emcasa/ui-dom/components/Button'
import Link from './components/Link'

const ROUTE_HOME = '/'
const ROUTE_MY_LISTINGS = '/listings/user-listings'

class Success extends PureComponent {
  constructor(props) {
    super(props)
    this.viewMyListings = this.viewMyListings.bind(this)
    this.goHome = this.goHome.bind(this)
  }

  componentDidMount() {
    routerEvents.once('routeChangeComplete', () => {
      this.props.navigateTo('intro')
      this.props.resetStore()      
      return true
    })
  }

  viewMyListings() {
    Router.push(ROUTE_MY_LISTINGS)
  }

  goHome() {
    Router.push(ROUTE_HOME)
  }

  render() {
    const listingId = this.props.listing.id
    return (
      <div ref={this.props.hostRef}>
        <Row justifyContent="center" p={4}>
          <Col width={[1, 1/2]}>
            <Text
              fontSize="large"
              fontWeight="bold"
              textAlign="center">
              Parabéns, seu imóvel foi salvo com sucesso!
            </Text>
            <Row justifyContent="center" my={4}>
              <Icon name="check-circle" color="green" size={60} />
            </Row>
            <Row>
              <Text color="grey">Muito obrigado por confiar na EmCasa! Enviaremos um SMS assim que seu imóvel estiver publicado.</Text>
            </Row>
            <Row>
              <Text color="grey">Você já tem fotos do seu imóvel? <Link
                href={`/listings/images?listingId=${listingId}`}
                as={`/imoveis/${listingId}/imagens`}
              >Clique aqui</Link> para fazer o upload.</Text>
            </Row>
            <Row>
              <Col width={1} mb={2}>
                <Button active fluid height="tall" onClick={this.viewMyListings}>Ver Meus Imóveis</Button>
              </Col>
            </Row>
            <Row>
              <Col width={1}>
                <Button fluid height="tall" onClick={this.goHome}>Ir para Home</Button>
              </Col>
            </Row>
          </Col>
        </Row>
      </div>
    )
  }
}

export default Success
