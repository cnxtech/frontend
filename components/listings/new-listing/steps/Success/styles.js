import styled from 'styled-components'
import {themeGet} from 'styled-system'
import Row from '@emcasa/ui-dom/components/Row'

export const Box = styled(Row)`
  border: 1px solid ${themeGet('colors.lightGrey')};
  box-shadow: 0px 2px 2px rgba(0, 0, 0, 0.3);
  border-radius: 2px;
`
