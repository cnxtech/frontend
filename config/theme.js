import theme from '@emcasa/ui'
import {fas} from '@fortawesome/pro-solid-svg-icons'
import {fal} from '@fortawesome/pro-light-svg-icons'
import icons from './icons'

export default {
  ...theme,
  icons: {emcasa: icons, default: fas, light: fal}
}
