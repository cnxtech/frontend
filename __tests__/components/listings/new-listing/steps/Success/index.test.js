import React from 'react';
import Success from 'components/listings/new-listing/steps/Success';
import renderer from 'react-test-renderer';
import 'jest-styled-components'
import theme from '@emcasa/ui'
import { ThemeProvider } from 'styled-components'

it('renders correctly', () => {
  const tree = renderer.create(
    <ThemeProvider theme={theme}>
      <Success
        tour={{
          day: '2018-11-23'
        }}
        location={{
          address: 'Av. Vieira Souto, 600'
        }}
        pricing={{
          userPrice: 1000000
        }}
        listing={{
          id: 1
        }}
      />
    </ThemeProvider>).toJSON()
  expect(tree).toMatchSnapshot()
})
