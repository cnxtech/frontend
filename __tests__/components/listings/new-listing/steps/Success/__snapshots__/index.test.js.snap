// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c8 {
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

.c9 {
  padding: 20px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 20px;
  padding-top: 0px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 20px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 10px;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c25 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 20px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c28 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 100%;
}

.c13 {
  width: 50%;
}

.c14 {
  margin-right: 2px;
  padding-top: 2px;
  width: 16.666666666666664%;
}

.c16 {
  width: 83.33333333333334%;
}

.c31 {
  margin-bottom: 10px;
  width: 100%;
}

.c33 {
  width: 100%;
}

.c2 {
  -webkit-font-smoothing: antialiased;
  font-family: FaktSoftPro-Normal;
  color: #3c4858;
  font-size: 24px;
  font-weight: bold;
  text-align: center;
}

.c4 {
  -webkit-font-smoothing: antialiased;
  font-family: FaktSoftPro-Normal;
  color: #979797;
  font-size: 18px;
}

.c10 {
  -webkit-font-smoothing: antialiased;
  font-family: FaktSoftPro-Normal;
  color: #979797;
  font-size: 14px;
}

.c12 {
  -webkit-font-smoothing: antialiased;
  font-family: FaktSoftPro-Normal;
  color: #3c4858;
  font-size: 14px;
  font-weight: bold;
}

.c15 {
  -webkit-font-smoothing: antialiased;
  font-family: FaktSoftPro-Normal;
  color: #3c4858;
  font-size: 12px;
}

.c27 {
  -webkit-font-smoothing: antialiased;
  font-family: FaktSoftPro-Normal;
  color: #f50057;
  font-size: 18px;
}

.c30 {
  -webkit-font-smoothing: antialiased;
  font-family: FaktSoftPro-Normal;
  color: #f50057;
  font-size: 14px;
}

.c32 {
  outline: none;
  display: block;
  cursor: pointer;
  padding: 0 20px 0 20px;
  border-color: #f50057;
  background-color: #f50057;
  border-radius: 4px;
  border-width: 1px;
  width: 100%;
  height: 60px;
  color: #ffffff;
  font-family: FaktSoftPro-Normal;
  font-size: 18px;
  -webkit-font-smoothing: antialiased;
  border-style: solid;
}

.c32 :hover {
  background-color: #f50057;
}

.c34 {
  outline: none;
  display: block;
  cursor: pointer;
  padding: 0 20px 0 20px;
  border-color: #e0e6ed;
  background-color: #ffffff;
  border-radius: 4px;
  border-width: 1px;
  width: 100%;
  height: 60px;
  color: #3c4858;
  font-family: FaktSoftPro-Normal;
  font-size: 18px;
  -webkit-font-smoothing: antialiased;
  border-style: solid;
}

.c34 :hover {
  background-color: #ffffff;
}

.c29 {
  margin-right: 10px;
}

.c26 {
  cursor: pointer;
}

.c6 {
  min-width: 300px;
  max-width: 300px;
}

.c22 {
  height: 1px;
  width: 220px;
  background-color: #e0e6ed;
}

.c19 {
  margin-top: -2px;
  margin-left: 4px;
  width: 10px;
  height: 10px;
  border-top: 1px solid rgba(0,0,0,0.4);
  border-radius: 0 100% 0 0;
  box-shadow: rgba(0,0,0,0.2) -1px 1px 2px 0px inset;
  background-color: #fff;
}

.c23 {
  margin-left: 4px;
  width: 10px;
  height: 10px;
  border-bottom: 1px solid rgba(0,0,0,0.2);
  border-radius: 0 0 100% 0;
  box-shadow: rgba(0,0,0,0.2) -1px -1px 1px 0px inset;
  background-color: #fff;
}

.c20 {
  margin-top: -2px;
  margin-right: 4px;
  width: 10px;
  height: 10px;
  border-top: 1px solid rgba(0,0,0,0.4);
  border-radius: 100% 0 0 0;
  box-shadow: rgba(0,0,0,0.2) 1px 1px 2px 0px inset;
  background-color: #fff;
}

.c24 {
  margin-right: 4px;
  width: 10px;
  height: 10px;
  border-bottom: 1px solid rgba(0,0,0,0.2);
  border-radius: 0 0 0 100%;
  box-shadow: rgba(0,0,0,0.2) 1px -1px 2px 0px inset;
  background-color: #fff;
}

.c21 {
  position: absolute;
  margin-left: 45px;
}

.c17 {
  background-color: #fff;
  width: calc(100% + 10px);
  margin-left: -5px;
}

@media screen and (min-width:780px) {
  .c1 {
    width: 50%;
  }
}

<div>
  <div
    className="c0"
  >
    <div
      className="c1"
      width={
        Array [
          1,
          0.5,
        ]
      }
    >
      <p
        className="c2"
        color="dark"
        fontSize="large"
        fontWeight="bold"
      >
        Parab√©ns, seu im√≥vel foi salvo com sucesso!
      </p>
      <div
        className="c3"
      >
        <p
          className="c4"
          color="grey"
          fontSize="default"
        >
          Muito obrigado por confiar na EmCasa!
           Aguarde o contato do nosso time por telefone em breve para confirmar a visita.
        </p>
      </div>
      <div
        className="c5"
      >
        <div
          className="c6 c3"
        >
          <div
            className="c7"
          >
            <div
              className="c8"
              elevation={4}
              style={
                Object {
                  "borderRadius": 4,
                }
              }
            >
              <div
                className="c9"
              >
                <div
                  className="c3"
                >
                  <div
                    className=""
                  >
                    <span
                      className="c10"
                      color="grey"
                      fontSize="small"
                    >
                      Endere√ßo
                    </span>
                  </div>
                </div>
                <div
                  className="c11"
                >
                  <div
                    className=""
                  >
                    <span
                      className="c12"
                      color="dark"
                      fontSize="small"
                      fontWeight="bold"
                    >
                      Av. Vieira Souto, 600
                    </span>
                  </div>
                </div>
                <div
                  className="c3"
                >
                  <div
                    className="c13"
                    width={0.5}
                  >
                    <span
                      className="c10"
                      color="grey"
                      fontSize="small"
                    >
                      Valor
                    </span>
                  </div>
                  <div
                    className="c13"
                    width={0.5}
                  >
                    <span
                      className="c10"
                      color="grey"
                      fontSize="small"
                    >
                      Data para visita
                    </span>
                  </div>
                </div>
                <div
                  className="c3"
                >
                  <div
                    className="c13"
                    width={0.5}
                  >
                    <span
                      className="c12"
                      color="dark"
                      fontSize="small"
                      fontWeight="bold"
                    >
                      R$1,000,000
                    </span>
                  </div>
                  <div
                    className="c13"
                    width={0.5}
                  >
                    <span
                      className="c12"
                      color="dark"
                      fontSize="small"
                      fontWeight="bold"
                    >
                      00/00/0000 - entre 00h e 00h
                    </span>
                  </div>
                </div>
                <div
                  className="c3"
                >
                  <div
                    className="c13"
                    width={0.5}
                  >
                     
                  </div>
                  <div
                    className="c13"
                    width={0.5}
                  >
                    <div
                      className="c3"
                    >
                      <div
                        className="c14"
                        width={0.16666666666666666}
                      >
                        <span
                          className="c15"
                          color="dark"
                          fontSize={12}
                        >
                          üìû
                        </span>
                      </div>
                      <div
                        className="c16"
                        width={0.8333333333333334}
                      >
                        <span
                          className="c10"
                          color="grey"
                          fontSize="small"
                        >
                           Aguarde a confirma√ß√£o da visita.
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="c17 "
              >
                <div
                  className="c18"
                >
                  <div
                    className="c19"
                  />
                  <div
                    className="c20"
                  />
                </div>
                <div
                  className="c21 c3"
                >
                  <div
                    className="c22 "
                  />
                </div>
                <div
                  className="c18"
                >
                  <div
                    className="c23"
                  />
                  <div
                    className="c24"
                  />
                </div>
              </div>
              <div
                className="c25"
              >
                <span
                  className="c26 c27"
                  color="pink"
                  fontSize="default"
                  onClick={[Function]}
                >
                  <a>
                    <div
                      className="c28"
                    >
                      <div
                        className="c29"
                      >
                        <svg
                          height={16}
                          version="1.1"
                          viewBox="0 0 448 512"
                          x={0}
                          y={0}
                        >
                          <path
                            d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                            fill="#f50057"
                            mr={2}
                            name="plus"
                            type="default"
                          />
                        </svg>
                      </div>
                      <span
                        className="c30"
                        color="pink"
                        fontSize="small"
                      >
                        Adicionar fotos
                      </span>
                    </div>
                  </a>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="c3"
      >
        <div
          className="c31"
          width={1}
        >
          <button
            className="c32"
            fontSize="default"
            height="tall"
            onClick={[Function]}
          >
            Ver Meus Im√≥veis
          </button>
        </div>
      </div>
      <div
        className="c3"
      >
        <div
          className="c33"
          width={1}
        >
          <button
            className="c34"
            fontSize="default"
            height="tall"
            onClick={[Function]}
          >
            Fale com a EmCasa
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
